src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -968,20 +968,31 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 
                 auto clickCount = _NumberOfClicks(cursorPosition, point.Timestamp());
 
+                // performs a double/triple click selection
+                auto multiClickSelection = [this, terminalPosition, shiftEnabled](::Terminal::SelectionExpansionMode mode) {

```

Comment:
i mean, I was 100% thinking more like...

inside each branch of the if, we just set an `std::optional` for the mode, and if it's set later we just "if shift { do thing A(mode) } else {do thing B(mode)}"

Don't over-complicate things!